dymos/trajectory/test/test_trajectory_parameters.py

```diff
@@ -0,0 +1,330 @@
+                          rate_continuity=True, rate_continuity_scaler=100.0,
+                          rate2_continuity=False)
+
+        # Unit introspection for phase params. This works.
+        phase.add_parameter('S', val=49.2386)
+
+        # Unit introspection for traj params. This doesn't work.
+        traj.add_parameter('Isp', val=1600.0, targets={'phase0': None})
+
+        phase.add_parameter('throttle', val=1.0, opt=False, targets=['throttle'])
+
+        with self.assertRaises(ValueError) as e:
+            p.setup(check=True)
+
+        expected = 'No target was found for trajectory parameter `Isp` in any phase.\n' \
+                   'Option `targets` is a dictionary keyed by phase name but target for each phase is None.'
+
+        self.assertEqual(str(e.exception), expected)
+
+    def test_parameter_introspection_targets_dict_no_valid_parameter_targets(self):
+        import openmdao.api as om
+
+        import dymos as dm
+
+        from dymos.examples.min_time_climb.min_time_climb_ode import MinTimeClimbODE
+
+        p = om.Problem()
+
+        traj = dm.Trajectory()
+
+        phase = dm.Phase(ode_class=MinTimeClimbODE,
+                         transcription=dm.GaussLobatto(num_segments=5, compressed=False))
+
+        traj.add_phase('phase0', phase)
+
+        p.model.add_subsystem('traj', traj)
+
+        phase.set_time_options(fix_initial=True, duration_bounds=(50, 400),
+                               duration_ref=100.0)
+
+        phase.add_state('r', fix_initial=True, lower=0, upper=1.0E6, units='m',
+                        ref=1.0E3, defect_ref=1.0E3,
+                        rate_source='flight_dynamics.r_dot')
+
+        phase.add_state('h', fix_initial=True, lower=0, upper=20000.0, units='m',
+                        ref=100.0, defect_ref=1.0E2,
+                        rate_source='flight_dynamics.h_dot')
+
+        phase.add_state('v', fix_initial=True, lower=10.0, units='m/s',
+                        ref=1.0E2, defect_ref=1.0E2,
+                        rate_source='flight_dynamics.v_dot')
+
+        phase.add_state('gam', fix_initial=True, lower=-1.5, upper=1.5, units='rad',
+                        ref=1.0, defect_ref=1.0,
+                        rate_source='flight_dynamics.gam_dot')
+
+        phase.add_state('m', fix_initial=True, lower=10.0, upper=1.0E5, units='kg',
+                        ref=1.0E3, defect_ref=1.0E3,
+                        rate_source='prop.m_dot')
+
+        phase.add_control('alpha', units='deg', lower=-8.0, upper=8.0, scaler=1.0,
+                          rate_continuity=True, rate_continuity_scaler=100.0,
+                          rate2_continuity=False)
+
+        # Unit introspection for phase params. This works.
+        phase.add_parameter('S', val=49.2386)
+
+        # Unit introspection for traj params. This doesn't work.
+        traj.add_parameter('Isp', val=1600.0, targets={'phase0': 'Isp'})
```

Comment:
I'm confused that this one raises an error message.  Isp is a variable in the phase (though not a parameter.)  BTW, I ran my LEAPS stuff with your branch, and it works. This is my add_parameter:

    for name in aero_inputs:
        traj.add_parameter(name, units=MetaData[name]['units'],
                           targets={'climb': [name]}, shape=1, static_target=True)

I guess it only works for me because static_target is True, but I am not sure why that makes a difference here.

Reply:
So it's a very subtle difference, which is why I've come to prefer forcing users to explicitly add parameters to phases and then let them be targed by trajectory parameters.

In `targets={'phase0': 'Isp'}`, `'Isp'` is a string so it's assumed to be a parameter.  If `'Isp'` was provided in a sequence (`['Isp']`), then it would assume its to find it in the ODE.  Some of this implementation dates back to the time before we could do introspection.  

I think things are a lot simpler with being explicit here:  Parameters provide an input to get external values into the phase, and we can use trajectory parameters to connect an external output to a single target (the trajectory parameter) and have connect to all relevant parameters in the phase.

I'd be in favor of ONLY documenting that way, and deprecating the use of a sequence for targets.